app/frontend/wallet/account.ts

```diff
@@ -294,30 +295,43 @@ const Account = ({
     return _getMaxSendableAmount(utxos, address, sendAmount)
   }
 
-  const selectUTxOs = async (txType: TxType): Promise<UTxO[]> => {
-    const availableUtxos = await getUtxos()
-    const nonStakingUtxos = availableUtxos.filter(({address}) => !isBase(addressToHex(address)))
-    const baseAddressUtxos = availableUtxos.filter(({address}) => isBase(addressToHex(address)))
+  const selectUTxOs = (utxos: UTxO[], txPlanArgs: TxPlanArgs): UTxO[] => {
+    const nonStakingUtxos = utxos.filter(({address}) => !isBase(addressToHex(address)))
+    const baseAddressUtxos = utxos.filter(({address}) => isBase(addressToHex(address)))
+    const adaOnlyUtxos = baseAddressUtxos.filter(({tokenBundle}) => tokenBundle.length === 0)
+    const tokenUtxos = baseAddressUtxos.filter(({tokenBundle}) => tokenBundle.length > 0)
     const randomGenerator = PseudoRandom(seeds.randomInputSeed)
-    // TODO: ideally we want to do nonStaking, with token, only ada, the rest
-    if (txType === TxType.SEND_ADA || txType === TxType.CONVERT_LEGACY) {
+
+    if (
+      txPlanArgs.txType === TxType.SEND_ADA &&
+      txPlanArgs.sendAmount.assetFamily === AssetFamily.TOKEN
+    ) {
+      const {policyId, assetName} = txPlanArgs.sendAmount.token
+      const utxosWithToken = tokenUtxos.filter(({tokenBundle}) =>
```

Comment:
This naming is possibly confusing as both the utxos "with" and "without" the token actually contain some token. Therefore I'd name this `utxosWithTargetToken` and `utxosWithoutTargetToken`

Reply:
agreed, thanks :+1: